pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/SchemasResourceBase.java

```diff
@@ -117,7 +117,7 @@ public void deleteSchema(boolean authoritative, AsyncResponse response) {
         validateDestinationAndAdminOperation(authoritative);
 
         String schemaId = getSchemaId();
-        pulsar().getSchemaRegistryService().deleteSchema(schemaId, defaultIfEmpty(clientAppId(), ""))
+        pulsar().getSchemaRegistryService().putEmptySchema(schemaId, defaultIfEmpty(clientAppId(), ""))
```

Comment:
Here it is probably more appropriate to call `deleteSchemaFromStorage()`

Reply:
as per the current code, when user tries to explicitly delete a schema from the topic, an empty schema is pushed to the schema storage so that the latest schema is an empty schema. this preserves the older versions of the schema in the storage. the topic itself remains unchanged.
the schema is deleted from storage only when the topic itself is deleted.

i expected that this would affect the behavior of a transitive compatibility check. however, that's not the case. the schema registry contains additional logic to ignore all existing schema versions if the latest schema instance is empty. by removing `deleteSchema(...)` and replacing its usages with `deleteSchemaStorage(...)` we can remove the `trimDeletedSchemaAndGetList(...)` method as well from the schema registry.
what do you think?